bin/node/runtime/src/lib.rs

```diff
@@ -2073,6 +2074,10 @@ impl_runtime_apis! {
 			let nodes = leaves.into_iter().map(|leaf|mmr::DataOrHash::Data(leaf.into_opaque_leaf())).collect();
 			pallet_mmr::verify_leaves_proof::<mmr::Hashing, _>(root, nodes, proof)
 		}
+
+		fn block_num_to_leaf_index(block_num: &BlockNumber) -> Result<mmr::LeafIndex, mmr::Error> {
+			Ok(Mmr::block_num_to_leaf_index(*block_num))
+		}
```

Comment:
Instead of this new translation API function, I was thinking of actually changing existing API functions to have `BlockNumber` params type instead of `LeafIndex`, but let's see what feedback we get on https://github.com/paritytech/substrate/issues/12339

Reply:
Well, that is actually what I did. I changed the argument in the existing RPC functions from `LeafIndex` to `BlockNumber`. I don't think we could have the translation logic inside these RPC functions cause we need access to the leaf count that is stored in the pallet storage and also need to get the current block number. So it made more sense to me to have this logic inside the pallet, and make it accessible from the runtime api.

Reply:
Sorry, I didn't read the message correctly 😆